setuptools/_distutils/unixccompiler.py

```diff
@@ -145,6 +145,10 @@ class UnixCCompiler(CCompiler):
     if sys.platform == "cygwin":
         exe_extension = ".exe"
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.compile_commands = []
```

Comment:
Let's see if we can figure out an integration that doesn't require multiple touch points, maybe by wrapping spawn. Ideally, the implementation would be one line added somewhere to the existing implementation for integration and the rest of the behavior would be encapsulated in a function or class.